bluetooth/src/commonMain/kotlin/scanner/Scanner.kt

```diff
@@ -79,26 +80,34 @@ abstract class BaseScanner constructor(
     open fun startMonitoringPermissions() {
         if (monitoringPermissionsJob != null) return
         monitoringPermissionsJob = launch(stateRepo.coroutineContext) {
-            bluetoothPermissionRepo.collect { state ->
-                handlePermissionState(state, BluetoothPermission)
+            permissionsFlow.collect { state ->
+                handlePermissionState(state)
             }
         }
     }
 
-    protected suspend fun <P : Permission> handlePermissionState(state: PermissionState<P>, permission: P) {
-        when (state) {
-            is PermissionState.Denied.Requestable -> if (autoRequestPermission) state.request(permissions.getManager(permission))
-            else -> {}
+    private suspend fun handlePermissionState(states: List<PermissionState<*>>) {
+        if (autoRequestPermission) {
+            states.forEach { state ->
+                when (state) {
+                    is PermissionState.Denied.Requestable -> state.request()
+                    else -> {}
+                }
```

Comment:
I know it's a copy/paste but this is a weird way to write `if (state is PermissionState.Denied.Requestable) state.request()`

Reply:
Fixing this in BT rewrite